shelley/chain-and-ledger/executable-spec/test/Generator/Block.hs

```diff
@@ -93,7 +96,7 @@ genBlock sNow chainSt coreNodeKeys keysByStakeHash = do
   lookForPraosStart <- genSlotIncrease
   let poolParams = (Map.toList . unPoolDistr . nesPd . chainNes) chainSt
   poolParams' <- take 1 <$> QC.shuffle poolParams
-  let (nextSlot, keys) = case poolParams' of
+  let (_nextSlot, _keys) = case poolParams' of
```

Comment:
why the move from `nextSlot` to `nextOSlot`? in case it was not clear, `nextOSlot` is the next overlay slot, not necessarily the slot we want to choose next, for we may instead pick a praos slot.

Reply:
the problem is that we currently don't seem to correctly select the next Praos slot, in particular the vrfcheck is not taken into account

Reply:
we are constructing a block with a leader nonce value of 0, so I guess the VRF witness for either the leader nonce or the block nonce are invalid?

Reply:
is the problem that `mkBlock` does not construct the correct VRF witnesses when crossing the epoch boundary?

Reply:
if you want I can did deeper into this, I had tried it a got a slot leader with `0 % 1` stake right away with the corresponding predicate failure `NotPraosLeaderOVERLAY`. My suggestion would be to merge this PR and do another PR that includes this.
With 0 the check results in 0 < 0 , doesn't it ? Maybe that is the failure in that case?

Reply:
that makes sense. I'm happy merging this PR now and getting the decentralized block put back in soon.